lib/tasks/scripts/telemedicine_reports.rb

```diff
@@ -0,0 +1,322 @@
+# frozen_string_literal: true
+
+require "csv"
+
+module TelemedicineReports
+  class << self
+    def parse_mixpanel(mixpanel_csv_path)
+      mixpanel_csv = File.read(mixpanel_csv_path)
+      mixpanel_data = CSV.parse(mixpanel_csv, headers: false)
+
+      mixpanel_data.drop(1).map { |row|
+        facility = User.find(row[1]).registration_facility
+        {user_id: row[1],
+         date: Date.parse(row[2]),
+         clicks: row[3].to_i,
+         facility_id: facility.id,
+         district: facility.district,
+         state: facility.state,
+         type: facility.facility_type}
+      }
+    end
+
+    def generate_report(mixpanel_data, period_start, period_end)
+      raw_mixpanel_data = mixpanel_data.select { |row|
+        (row[:date] >= period_start) && (row[:date] <= period_end)
+      }
+      mixpanel_data = format_mixpanel_data(raw_mixpanel_data)
+
+      facilities = Facility.where(enable_teleconsultation: true).map { |facility|
```

Comment:
We only want detailed reporting for HWCs and SCs where teleconsultation is enabled